crates/extensions/c8y_firmware_manager/src/worker.rs

```diff
@@ -371,10 +373,8 @@ impl FirmwareManagerWorker {
 
         let c8y_child_topic = C8yTopic::ChildSmartRestResponse(child_id.to_string())
             .to_topic(&self.config.c8y_prefix)?;
-        let executing_msg = MqttMessage::new(
-            &c8y_child_topic,
-            DownloadFirmwareStatusMessage::status_executing(),
-        );
+        let payload = set_operation_executing_with_name(CumulocitySupportedOperations::C8yFirmware);
```

Comment:
thought: wasn't aware that firmware manager doesn't go through the mapper but publishes c8y messages directly
Are there any plans to have it handled by the mapper or does the firmware manager benefit from having a direct line to c8y without going through the mapper?

Reply:
This `c8y_firmware_manager` is used only for the legacy `c8y_firmware_plugin`. All legacy operation plugins have direct MQTT connection without mapper in order to delivery the feature faster.

Reply:
Technically, we don't have to update such legacy code. However, I did here because I wanted to get rid of `OperationStatusMessage`. To avoid someone will use the duplicated old API to update operation status.